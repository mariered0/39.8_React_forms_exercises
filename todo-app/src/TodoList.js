import React from 'react';
import NewTodoForm from './NewTodoForm';

const TodoList = () => {

    return (
        <div className='TodoList'>
            <h1>TodoList</h1>
            <NewTodoForm />
        </div>
    )
}

export default TodoList;