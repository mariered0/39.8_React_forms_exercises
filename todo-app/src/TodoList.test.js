import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

function addTodo (todoList, todo="Buy milk") {
    const inputTodo = todoList.getByLabelText("Todo:");
    const btn = todoList.queryByText("Add");
    fireEvent.change(inputTodo, {target: {value: todo}});
    fireEvent.click(btn);
}

it("renders without crashing", function() {
    render(<TodoList />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it("should display the title", function () {
    const { queryByText } = render(<TodoList />);
    expect(queryByText("TodoList")).toBeInTheDocument();
})

it("can add a new todo", function () {
    const todoList = render(<TodoList />);

    //there's onetodo displayed by default
    expect(todoList.queryByText("x")).toBeInTheDocument();

    addTodo(todoList);

    const xButton = todoList.queryAllByText("x");
    expect(xButton.length).toBe(2);
    expect(todoList.queryByText("Buy milk")).toBeInTheDocument();
})