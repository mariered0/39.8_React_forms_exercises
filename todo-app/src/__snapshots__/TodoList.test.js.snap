// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList-container"
  >
    <h1>
      TodoList
    </h1>
    <form>
      <label
        for="todo"
      >
        Todo: 
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="todo"
        type="text"
        value=""
      />
      <button>
        Add
      </button>
    </form>
    <div>
      <ul>
        <div
          class="Todo-container"
        >
          <li
            class="Todo"
            style="list-style-type: none;"
          >
            Feed cats
          </li>
          <button
            class="Todo-delete-btn"
            id="2a96bc22-34ce-4e73-bb1f-c362d7d6569c"
          >
            x
          </button>
        </div>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
